integration_test/tests/advertisingTest.ts

```diff
@@ -0,0 +1,181 @@
+import { TestScope } from 'cavy';
+import {
+  callPlayer,
+  callPlayerAndExpectEvent,
+  callPlayerAndExpectEvents,
+  EventSequence,
+  EventType,
+  expectEvent,
+  expectEvents,
+  loadSourceConfig,
+  RepeatedEvent,
+  startPlayerTest,
+} from '../playertesting';
+import { Sources } from './helper/Sources';
+import {
+  AdItem,
+  AdSourceType,
+  AdvertisingConfig,
+} from 'bitmovin-player-react-native';
+import { AdTags } from './helper/Ads';
+import { Platform } from 'react-native';
+
+export default (spec: TestScope) => {
+  function commonAdvertisingTests(
+    adItems: AdItem[],
+    type: 'VAST' | 'VMAP' | 'progressive'
+  ) {
+    spec.describe(`scheduling ${type} ads via AdvertisingConfig`, () => {
+      spec.it('emits AdScheduled events', async () => {
+        const advertisingConfig = {
+          schedule: adItems,
+        } as AdvertisingConfig;
+        await startPlayerTest({ advertisingConfig }, async () => {
+          await callPlayerAndExpectEvents((player) => {
+            player.load(Sources.artOfMotionHls);
+          }, RepeatedEvent(EventType.AdScheduled, adItems.length));
+        });
+      });
+    });
+    spec.describe(`scheduling ${type} ads via scheduleAd API`, () => {
+      spec.it('emits AdScheduled events', async () => {
+        await startPlayerTest({}, async () => {
+          await loadSourceConfig(Sources.artOfMotionHls);
+          await callPlayerAndExpectEvents((player) => {
+            adItems.forEach((adItem) => player.scheduleAd(adItem));
+          }, RepeatedEvent(EventType.AdScheduled, adItems.length));
+        });
+      });
+    });
+    spec.describe(`playing ${type} ads`, () => {
+      spec.it('emits Ad events', async () => {
+        const advertisingConfig = {
+          schedule: adItems,
+        } as AdvertisingConfig;
+        await startPlayerTest({ advertisingConfig }, async () => {
+          await callPlayer(async (player) => {
+            player.load(Sources.artOfMotionHls);
+            player.play();
+          });
+          if (type !== 'progressive') {
+            await expectEvent(EventType.AdManifestLoaded, 20);
+          }
```

Comment:
I experienced longer waiting times with the Android emulator, therefore added this extra expectation to stabilize the test. APPLIES TO THE OTHER OCCURRENCES TOO